frodo.js

```diff
@@ -1,132 +1,130 @@
 /**
```

Comment:
SCRIPT-This new feature lacks unit tests. Could you add some to ensure everything works correctly?